components/elm/src/data_types/MOABGridType.F90

```diff
@@ -528,6 +532,9 @@ subroutine read_grid_cell_lat_lon()
     allocate(moab_gcell%lonv(moab_gcell%num_ghosted, moab_gcell%nv))
     allocate(data2d(moab_gcell%nv, moab_gcell%num_ghosted))
```

Comment:
@bishtgautam Are these allocated structures deallocated somewhere? I do not see one for `moab_gcell%lonv, moab_gcell%latv, data2d, moab_gcell%is_vert_duplicate` variables from a cursory look.